pkg/grizzly/workflow.go

```diff
@@ -316,6 +316,40 @@ func Watch(config Config, watchDir string, parser Parser) error {
 	return nil
 }
 
+// WatchResource waits for remote changes to a resource and saves them to disk
+func WatchResource(config Config, UID, filename string) error {
+	count := strings.Count(UID, ".")
+	var handlerName, resourceID string
+	if count == 1 {
+		parts := strings.SplitN(UID, ".", 2)
+		handlerName = parts[0]
+		resourceID = parts[1]
+	} else if count == 2 {
+		parts := strings.SplitN(UID, ".", 3)
+		handlerName = parts[0] + "." + parts[1]
+		resourceID = parts[2]
+
+	} else {
+		return fmt.Errorf("UID must be <provider>.<uid>: %s", UID)
```

Comment:
This seems inconsistent with the two possible branches above - could be two OR three parts. Is this an artifact of some old form of our dashboard UIDs?

Reply:
Before when Grizzly just handled dashboards, the UID was just the dashboard UID. Now we need to 'namespace' the UID, so we might call it `dashboard.my-dash`. But what if Grizzly might support more than one dashboard provider? Then we would prefix it with a provider name: `grafana.dashboard.my-dash`. That's the idea, anyway. Maybe overkill. And the error is unhelpful and out of date - it is really the 'object type' or 'kind' or such, not the 'provider'. I need to think about that.